### PR TITLE
[jeelink] Add support for sensors with multiple channels.

### LINKED ISSUE #3027: [jeelink-lacrosse] Channel is not decoded
Ther are LaCrosse sensors (like TFA 30.314), that provide two temperatures using two channels. 
org.openhab.binding.jeelink dose not take care about this information. Therfore sensors like TFA 30.314 are showing strange behavior. 

The information about the channel is combined with the type. I think the second bit is the channel information. 

Here two readings from my TFA 30.314:
OK 9 59 130 4 56 125
OK 9 59 1 4 158 106
The correct representation of this is:
tempreture2:  8.0  °C
tempreture1: 18.2 °C
(don't now about the humidity values, since the sensor is temperature only)

### PR DESCRIPTION
**Pull request by vbier**
This adds support for LaCrosse sensors having more than one temperature/humidity sensor.

It dynamically adds channels to the Sensor thing for the additional sensors.

Closes #3027.

Signed-off-by: Volker Bier volker.bier@web.de

### CODE REVIEW

File: addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/internal/lacrosse/LaCrosseTemperatureSensorHandler.java [hunk 1]
```diff
@@ -44,20 +53,86 @@ public String getSketchName() {
 
     @Override
     public ReadingPublisher<LaCrosseTemperatureReading> createPublisher() {
+        return new ReadingPublisher<LaCrosseTemperatureReading>() {
+            private final HashMap<Integer, ReadingPublisher<LaCrosseTemperatureReading>> channelPublishers = new HashMap<>();
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/internal/lacrosse/LaCrosseTemperatureSensorHandler.java]: Please consider using the type for the field, so use Map there

**Comment by martinvw** [comment]
Thanks

**Comment by christianbur** [comment]
Hello, i habe a LaCrosse/Technoline TX25-IT with two temperature values.
what must be done so that the second temperature is processed as temperature and not as humidity?

**Comment by vbier** [comment]
Are you saying you get a second set of temperature and humidity channels and the second humidity channel shows the temperature?

Enable trace logging and post some of the readings you get from the sensor fro the log file.

**Comment by christianbur** [comment]
sorry i only got to know openhab last night, so i don't know how to activate the debug log.
But I'll give you some information, because I wrote a [program](https://github.com/christianbur/check_mk/blob/master/jeelink_TX29DTH/jeelinkD.py) for it myself a few years ago. 

the TX25-IT is a "dual channel" sensor. 
TX29DTH sensors always transmit temperature and humidity in channel 1. The TX25-IT uses two channels under the same id. In the first channel, the first temperature is transmitted and in the second channel the second temperature is transmitted. the humidity in both channels is unknown (over 106).
Openhab finds the TX25-IT but I only see the temperature and humidity. 

And in my opinion there is also something wrong in the processing of the sensor values. The sensor delivers in channel 1: 7.9 C and in channel2: -17.9 C and in openhab 

```
2018-04-01 15:37:24.108 [.ItemChannelLinkAddedEvent] - Link 'LaCrosse_KS_Temperatur-jeelink:lacrosse:2:temperature' has been added.
2018-04-01 15:37:29.211 [.ItemChannelLinkAddedEvent] - Link 'LaCrosse_KS_Luftfeuchtigkeit-jeelink:lacrosse:2:humidity' has been added.
2018-04-01 15:38:11.589 [vent.ItemStateChangedEvent] - LaCrosse_KS_Temperatur changed from NULL to -2.8
2018-04-01 15:38:11.595 [vent.ItemStateChangedEvent] - LaCrosse_KS_Luftfeuchtigkeit changed from NULL to 114
2018-04-01 15:39:11.601 [vent.ItemStateChangedEvent] - LaCrosse_KS_Temperatur changed from -2.8 to -4.8
2018-04-01 15:39:11.603 [vent.ItemStateChangedEvent] - LaCrosse_KS_Luftfeuchtigkeit changed from 114 to 115
2018-04-01 15:40:11.597 [vent.ItemStateChangedEvent] - LaCrosse_KS_Temperatur changed from -4.8 to -3.5
2018-04-01 15:40:11.603 [vent.ItemStateChangedEvent] - LaCrosse_KS_Luftfeuchtigkeit changed from 115 to 114
2018-04-01 15:41:11.600 [vent.ItemStateChangedEvent] - LaCrosse_KS_Temperatur changed from -3.5 to -6.2
2018-04-01 15:41:11.605 [vent.ItemStateChangedEvent] - LaCrosse_KS_Luftfeuchtigkeit changed from 114 to 116
```

the temperature is not correct but the humidity is also calculated incorrectly. 
 An humidity of over 100 (or 106) means that the sensor does not transmit humidity, but the wrong value ist trasmitted to openhab, this is wrong. 

please take a closer look at this.
http://fredboboss.free.fr/articles/tx29.php/

**Pull request merged**